src/utils/augmentation.py

```diff
@@ -0,0 +1,152 @@
+                    char_with_diac += randBytes.decode("utf-16be")
+                symbols.append(char_with_diac)
+            else:
+                symbols.append(ch)
+        return "".join(symbols)
+
+
+class SpaceAugmentation(AugmentationWord):
+    """
+    Adds spaces to chars of the word with a given probability
+    """
+
+    def __init__(self, expected_number_of_chars: float):
+        self.expected_number_of_chars = expected_number_of_chars
+
+    def __call__(self, word: str) -> str:
+        symbols = []
+        proba_per_char = (
+            self.expected_number_of_chars / len(word) if len(word) >= self.expected_number_of_chars else 0.5
+        )
+        for ch in word:
+            replace = np.random.binomial(1, proba_per_char)
+            if replace:
+                symbols.append(" ")
+            symbols.append(ch)
+        return "".join(symbols)
+
+
+class AugmentationText:
+    """
+    Generates a noisy text with given parameters:
+    text - original text to which word augmentations are applied
+    proba_per_text - probability of noise for a given text;
+    EXPECTED_NUMBER_OF_WORDS - expected value (average) of words in every text that we want to make noisy
+    EXPECTED_NUMBER_OF_CHARS - expected value of chars in a word that we want to make noisy
+    MAX_COUNT_AUGM - maximum value of augmentations that can be applied to every word
+    """
+
+    def __init__(self, leet_symbols: dict, cluster_symbols: dict, proba_per_text: float):
+        diacritics = DiacriticsAugmentation(EXPECTED_NUMBER_OF_CHARS)
+        clusters = SimilarCharAugmentation(cluster_symbols, EXPECTED_NUMBER_OF_CHARS)
+        leet = SimilarCharAugmentation(leet_symbols, EXPECTED_NUMBER_OF_CHARS)
+        spaces = SpaceAugmentation(EXPECTED_NUMBER_OF_CHARS)
+        swap = SwapAugmentation()
+        self.augmentations = np.array([diacritics, clusters, leet, spaces, swap])
+        self.augmentations_probas = [0.3, 0.3, 0.3, 0.05, 0.05]  # sum must be equal to 1
+        self.proba_per_text = proba_per_text
+
+    def __call__(self, text: str) -> str:
+        need_to_replace = np.random.binomial(1, self.proba_per_text)
+        if not need_to_replace:
+            return text
+
+        words = text.split()
+        result = ""
+
+        proba_per_word = EXPECTED_NUMBER_OF_WORDS / len(words) if len(words) >= EXPECTED_NUMBER_OF_WORDS else 0.5
+        for word in words:
+            if len(word) == 0:
+                continue
+
+            replace = np.random.binomial(1, proba_per_word)
+            if replace:
+                number_augmentations = np.random.choice(range(1, MAX_COUNT_AUGM + 1))
+                random_augmentations = np.random.choice(
+                    self.augmentations, number_augmentations, p=self.augmentations_probas, replace=False
+                )
+                for augmentation in random_augmentations:
+                    word = augmentation(word)
+            result += word + " "
```

Comment:
I suggest to add `IdAugmentation` that doesn't apply any transformation. Its probability would be `1-sum(p)`.
And after that you can use choice to select augmentations: `random_augmentations = np.random.choise(self.augmentations, p=self.probas)`. `IdAugmentation` will occurs 0 to N times, so it is similar to current logic, but much elegant.